ending_location_visitor.py

```diff
@@ -1,32 +1,259 @@
+            # logging.debug('\tLines {} to {}'.format(node.lineno, node.end_lineno))
+            # logging.debug('\tCols {} to {}'.format(node.col_offset, node.end_col_offset))
+        return self.props_check
 
-def set_const(node: astroid.Const):
-    """Populate ending locations for Const nodes."""
-    node.end_lineno = node.lineno
-    node.end_col_offset = node.col_offset + len(str(node.value))
+    def type(self, node_type):
+        """Set the type of node that we are inspecting. The type of node is
+        compared against in a private method.
+        """
+        self.node_type = node_type
+        return self;
 
+    def reset(self):
+        """Reset between test methods.
+        """
+        # Keep a list rather than a set, because "too many" is wrong.
+        self.props_check = []
+        self.node_type = None
 
-if __name__ == '__main__':
-    # TODO: turn this into a proper test
-    ending_transformer = TransformVisitor()
 
-    with open('examples/ending_locations/const.py') as f:
-        content = f.read()
 
-    ending_transformer.register_transform(astroid.Const, set_const)
 
-    module = astroid.parse(content)
-    ending_transformer.visit(module)
+################################################################################
+class TestEndingLocation(unittest.TestCase):
+    """
+    Prove the `end_lineno`, `end_col_offset` properties are set on ast nodes.
 
-    ending_visitor = EndingVisitor()
+    Method explanations:
+    ending_transformer.visit(module)
+        Walk the given astroid *tree* and transform each encountered node
+        *** Only the nodes which have transforms registered will actually
+        be replaced or changed.
+    
     ending_visitor.visit(module)
+        Show that the transform worked.
+
+    ending_transformer.easy_register_transform(<astroid-node-name>),
+    i.e.
+    ending_transformer.register_transform(astroid.<Type>, <function>)
+        Register `transform(node)` function to be applied on the given
+        astroid's `node_class`. Internally, appends the function to a
+        dictionary in the TransformVisitor class. The keys are the node types,
+        each node must have been assigned/registered a transform function!!
+        The transform function may return a value which is then used to
+        substitute the original node in the tree.
+        params: node_class, transform function
+
+    We store the correct values as: 
+        [lineno, end_lineno, col_offset, end_col_offset]
+    """
+
+    @classmethod
+    def setUpClass(self):
+        """A class method called before tests in an individual class run.
+        setUpClass is called with the class as the only argument and must be
+        decorated as a classmethod():"""
+        
+        # A visitor to transform the nodes.
+        self.ending_transformer = TransformVisitor()
+
+        # A visitor to test the nodes property correctness.
+        self.ending_visitor = EndingVisitor()
```

Comment:
You don't need a separate class here. A more elegant approach is to use the `nodes_of_class` Node method (https://github.com/PyCQA/astroid/blob/master/astroid/node_classes.py#L407). I believe @hayleyZHL did this, so you can ask her about it as well.

In general, it seems like trying to get a `TransformVisitor` to return some function of all the nodes is not idiomatic.
I did look for a bit to try to get a generator on all the nodes, and didn't find it in astroid, and it might be handy to create just that at a later point, using the approach in `nodes_of_class`.